packages/mdc-linear-progress/_mixins.scss

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@import "@material/theme/functions";
+@import "@material/theme/mixins";
+
+@mixin mdc-linear-progress-bar-color($color) {
+  .mdc-linear-progress__bar-inner {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-linear-progress-buffer-color($color) {
+  $color-value-for-css: mdc-theme-prop-value($color);
+  $color-value-for-svg: mdc-linear-progress-str-replace_(#{$color-value-for-css}, "#", "%23");
```

Comment:
Where does %23 come from? Is this for the SVG optimization as commented below?

Reply:
That's what I originally assumed, but it turns out it's actually because the SVG is in a data URI, and `#` is a reserved character in URIs. Without escaping, `#0af` would be interpreted as a URL hash and the color wouldn't be displayed correctly. (I only know this because I didn't escape it at first and it broke in Firefox 😛 Chrome was more forgiving and rendered it correctly.)

According to [Wikipedia's article on URL encoding](https://en.wikipedia.org/wiki/Percent-encoding), the characters `(`, `)`, `,`, and ` ` (space) are also reserved, yet both Chrome and Firefox seem perfectly happy to render `rgba(1, 2, 3, .5)` without escaping. 🤷‍♂️ 

I added a comment above those lines explaining why it's being replaced. Does that seem clearer to you?